### PR TITLE
feat: [Week-2.2] Translation and Update of "TypeScript Advanced" Module

### PR DESCRIPTION
**Pull request by haslie22**
#### Title of Pull Request

Translation and Update of "TypeScript Advanced" Module

#### 🤔 This is a ...

- [ ] 🌟 New task
- [ ] 🌐 New module
- [ ] ⚙️ Update to an existing task
- [x] 🔧 Update to an existing module
- [ ] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- [x] Module description translated
- [x] Module materials revised and updated

#### Additional Information

Board task: [1429](https://github.com/orgs/rolling-scopes-school/projects/4/views/1?pane=issue&itemId=48497396)

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [ ] 📝 I have commented my code, particularly in hard-to-understand areas.
- [ ] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

File: stage2/modules/typescript-advanced/README.md [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
 
-## Дополнительные материалы:
+## Theory 📖
+
+Students are encouraged to study the following resources:
+
+- [Learn Typescript Tutorial](https://scrimba.com/learn/typescript) - [3 hours]
+- [Utility Types - Advanced TypeScript](https://youtu.be/Fgcu_iB2X04?si=KzVUMpuuhnH92gRj) - [1 hour]
+- [Webpack & TypeScript Setup](https://youtube.com/playlist?list=PL4cUxeGkcC9hOkGbwzgYFmaxB0WiduYJC) - [1 hour]
+
```
Comment (valerydluski) [inline stage2/modules/typescript-advanced/README.md:26]:
```suggestion
- **[Type Guards]:**
   - [instanceof narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#instanceof-narrowing)
   - [typeof Operator](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#typeof-type-guards)
   - [Equality narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#equality-narrowing)
   - [Truthiness narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#truthiness-narrowing)
   - [Type Predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)
- **[Generics]:**
   - [Hello World of Generics](https://www.typescriptlang.org/docs/handbook/2/generics.html#hello-world-of-generics)
   - [Generic Constraints - TypeScript](https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-constraints)
- **[Utility Types](https://www.typescriptlang.org/docs/handbook/utility-types.html)**
- **Advanced Types:**
   - [Mapped Types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#handbook-content)
   - [Conditional Types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#handbook-content)
   - [Literal Types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types)
   - [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html#handbook-content)
```

File: stage2/modules/typescript-advanced/README.md [hunk 2]
```diff
@@ -1,29 +1,50 @@
 # [Stage#2.](../../) TypeScript Advanced
 
-## В данном модуле студентам необходимо приобрести следующие навыки:
+## Module Overview 📚
 
-- служебные типы для облегчения работы общих типов
-- настройка tsconfig
-- знания по настройке сборки с использованием TypeScript
+This module is designed for students to enhance their skills in TypeScript. It aims to provide an in-depth understanding of utility types for streamlined work with common types, configuring `tsconfig` effectively, and gaining knowledge in setting up builds using TypeScript.
```
Comment (valerydluski) [inline stage2/modules/typescript-advanced/README.md:5]: please update according to the links I wrote in the comments

**Review by valerydluski — COMMENTED** [review]
Great job, thank you

File: stage2/modules/typescript-advanced/README.md (hunk 2 above)
Comment (haslie22) [reply stage2/modules/typescript-advanced/README.md]: Done 👌

File: stage2/modules/typescript-advanced/README.md [hunk 3]
```diff
@@ -1,29 +1,67 @@
...
+- Complete the test "[St2] TS Advanced" in the RS-App > Auto Test.
+- Complete the task "[News API](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/migration-newip-to-ts.md)".
+
+## Additional Resources 📘
+
+Expand your knowledge with these additional materials:
+
+Theory:
 
 - [Utility Types](https://www.typescriptlang.org/docs/handbook/utility-types.html)
```
Comment (KristiBo) [inline stage2/modules/typescript-advanced/README.md:55]: Duplicates line 36

**Review by KristiBo — APPROVED** [review]

File: stage2/modules/typescript-advanced/README.md (hunk 3 above)
Comment (haslie22) [reply stage2/modules/typescript-advanced/README.md]: Fixed 👌

**Pull request merged**